.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,55 @@
+name: Build Status
+
+on:
+  push:
+    # Run tests on any push to feature branches.
+    branches: [ "master", "main", "DS-[0-9]+*", "ubid-test" ]
+  pull_request:
+    branches: [ "master", "main" ]
+
+jobs:
+  # Run an explicit Checkstyle stage to highlight any linting errors.
+  checkstyle:
+    name: mvn checkstyle
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install minimum-supported JDK for Checkstyle
+      uses: actions/setup-java@v3
+      with:
+        java-version: 11
+        distribution: 'temurin'
+        cache: maven
+
+    - name: Checkstyle
+      run: |
+        ./mvnw checkstyle:check -Dcheckstyle.config.location=config/scim2-parent-checkstyle.xml
```

Comment:
I added a [maven wrapper](https://maven.apache.org/wrapper/) so that we have firm control over the version of Maven that is used by these images.

Reply:
Seems like you have to add quite a lot to the repo (the jar file, etc) to get this wrapper. What's the cost of not doing this and just using some built in maven action?

Reply:
My concern about this is that I don't think there's a way to control the version of Maven that the pipeline pulls in. I could be wrong, but it would be suboptimal if the pipeline starts failing due to an incompatibility with a future version of the Maven client. For what it's worth, using Maven wrappers and committing them to source control is not uncommon.